debugger/src/debug_adapter/dap_adapter.rs

```diff
@@ -592,31 +557,26 @@ impl<P: ProtocolAdapter> DebugAdapter<P> {
             // The DAP Client will always do a `reset_and_halt`, and then will consider `halt_after_reset` value after the `configuration_done` request.
             // Otherwise the probe will run past the `main()` before the DAP Client has had a chance to set breakpoints in `main()`.
             match target_core.core.reset_and_halt(Duration::from_millis(500)) {
-                Ok(_) => {
+                Ok(core_info) => {
                     // Ensure ebreak enters debug mode, this is necessary for soft breakpoints to work on architectures like RISC-V.
                     target_core.core.debug_on_sw_breakpoint(true)?;
 
-                    if let Some(request) = request {
-                        return self.send_response::<()>(request, Ok(None));
-                    }
-                    // Only notify the DAP client if we are NOT in initialization stage (`CoreStatus::Unknown`).
-                    if self.last_known_status != CoreStatus::Unknown {
+                    // Only notify the DAP client if we are NOT in initialization stage ([`DebugAdapter::configuration_done`]).
+                    if self.configuration_done {
                         let event_body = Some(StoppedEventBody {
-                            reason: "reset".to_owned(),
```

Comment:
The DAP spec defines this as a string, and has some very specific requirements for this `reason` depending on the request it came from. There are literally 3 places (2 for 'restart', and 1 for 'pause')in the code where this has to be hardcoded as a string, so not sure it justifies an enum. Also, this will not proliferate over time, because of how tightly controlled the `StoppedEvent` handling needs to be.